### PR TITLE
addressing comments made about the maximum length of the keys and val…

### PR DESCRIPTION
**Pull request by westford14**
…ues and clarifying the docstrings

Signed-off-by: westford14 <westford14@gmail.com>

## What changes are proposed in this pull request?

This updates documentation to address issue #4202 

## How is this patch tested?

By visually inspecting the generated documentation

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by westford14** [comment]
hey @dbczumar -- sorry for switching to a new PR I had some issues with signoffs on the commits so i closed and reopened this PR

this should address your comment here https://github.com/mlflow/mlflow/pull/4965#pullrequestreview-794768159 about clarifying the maximum length vs what the backend stores can handle, also this PR should address the linting errors that were appearing

let me know what you think!

File: mlflow/tracking/_tracking_service/client.py [hunk 1]
```diff
@@ -248,8 +249,13 @@ def set_tag(self, run_id, key, value):
...
         :param run_id: String ID of the run.
-        :param key: Name of the tag.
-        :param value: Tag value (converted to a string)
+        :param key: Tag name (string). This string may only contain alphanumerics, underscores 
+                    (_), dashes (-), periods (.), spaces ( ), and slashes (/).
+                    All backend stores will support keys up to length 250 but some may 
+                    support larger key values.
+        :param value: Tag value (string, but will be string-ified if not).
+                      All backend stores will support values up to length 5000 but some 
+                      may support larger key values.
```
Comment (dbczumar) [inline mlflow/tracking/_tracking_service/client.py:258]:
```suggestion
                      All backend stores will support values up to length 5000, but some 
                      may support larger values.
```

File: mlflow/tracking/_tracking_service/client.py [hunk 2]
```diff
@@ -248,8 +249,13 @@ def set_tag(self, run_id, key, value):
         Set a tag on the run with the specified ID. Value is converted to a string.
 
         :param run_id: String ID of the run.
-        :param key: Name of the tag.
-        :param value: Tag value (converted to a string)
+        :param key: Tag name (string). This string may only contain alphanumerics, underscores 
+                    (_), dashes (-), periods (.), spaces ( ), and slashes (/).
+                    All backend stores will support keys up to length 250 but some may 
+                    support larger key values.
```
Comment (dbczumar) [inline mlflow/tracking/_tracking_service/client.py:255]:
```suggestion
                    All backend stores will support keys up to length 250, but some may 
                    support larger keys.
```

Comment (dbczumar) [reply mlflow/tracking/_tracking_service/client.py:255]: ^ Can we apply this change to the other relevant docstrings as well?

File: mlflow/tracking/_tracking_service/client.py (hunk 1 above)
Comment (dbczumar) [reply mlflow/tracking/_tracking_service/client.py:258]: ^ Can we apply this change to the other relevant docstrings as well?

File: mlflow/tracking/client.py [hunk 3]
```diff
@@ -636,7 +636,8 @@ def log_metric(
         Log a metric against the run ID.
 
         :param run_id: The run id to which the metric should be logged.
-        :param key: Metric name.
+        :param key: Metric name (string). This string may only contain alphanumerics, underscores
```
Comment (dbczumar) [inline mlflow/tracking/client.py:639]: Can we specify the key size that's guaranteed to be supported by all backends for metrics as well?

**Review by dbczumar — COMMENTED** [review]
@westford14 Thank you for pushing updates! I think this is almost ready - I just left a couple follow-up comments.

**Comment by westford14** [comment]
thanks for the comments! @dbczumar hopefully the most recent commit has addressed your previous comments

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @westford14 !

**Comment by dbczumar** [comment]
@westford14 Can you sign your commits using the instructions located here: https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#sign-your-work?

**Pull request closed without merging**